facebook-commerce.php

```diff
@@ -4240,49 +4240,57 @@ function fb_duplicate_product_reset_meta( $to_delete ) {
 
 	/**
 	 * Helper function to update FB visibility.
+	 *
+	 * @param int $product_id product ID
+	 * @param string $visibility visibility
 	 */
-	function update_fb_visibility( $wp_id, $visibility ) {
+	function update_fb_visibility( $product_id, $visibility ) {
 
 		// bail if the plugin is not configured properly
 		if ( ! $this->is_configured() || ! $this->get_product_catalog_id() ) {
 			return;
 		}
 
-		$woo_product = new WC_Facebook_Product( $wp_id );
+		$woo_product = new \WC_Facebook_Product( $product_id );
 
-		if ( ! $woo_product->exists() ) {
-			// This function can be called for non-woo products.
+		// bail if product isn't found
+		if ( ! $woo_product->woo_product instanceof \WC_Product || ! $woo_product->exists() ) {
 			return;
 		}
 
-		$products = WC_Facebookcommerce_Utils::get_product_array( $woo_product );
-		foreach ( $products as $item_id ) {
-			$fb_product_item_id = $this->get_product_fbid(
-				self::FB_PRODUCT_ITEM_ID,
-				$item_id
-			);
+		$should_set_visible = $visibility === self::FB_SHOP_PRODUCT_VISIBLE;
+
+		if ( ! $woo_product->woo_product->is_type( 'variable' ) ) {
+
+			$fb_product_item_id = $this->get_product_fbid( self::FB_PRODUCT_ITEM_ID, $product_id );
 
 			if ( ! $fb_product_item_id ) {
-				WC_Facebookcommerce_Utils::fblog(
-					$fb_product_item_id . " doesn't exist but underwent a visibility transform.",
-					array(),
-					true
-				);
-				  continue;
+				\WC_Facebookcommerce_Utils::fblog( $fb_product_item_id . " doesn't exist but underwent a visibility transform.", [], true );
+				 return;
 			}
-			$result = $this->check_api_result(
-				$this->fbgraph->update_product_item(
-					$fb_product_item_id,
-					array( 'visibility' => $visibility )
-				)
-			);
-			if ( $result ) {
 
-				$is_visible = $visibility === self::FB_SHOP_PRODUCT_VISIBLE;
+			$set_visibility = $this->fbgraph->update_product_item( $fb_product_item_id, [ 'visibility' => $visibility ] );
+
+			if ( $this->check_api_result( $set_visibility ) ) {
+				Products::set_product_visibility( $woo_product->woo_product, $should_set_visible );
+			}
+
+		} else {
+
+			$product_ids = array_merge( [ $product_id ], $woo_product->get_children() );
```

Comment:
If I'm not mistaken this would throw an exception now, as `$woo_product` is currently a `WC_Facebook_Product`.

Reply:
yeah one reason why I'd like to just use a plain \WC_Product; the facebook product does have a __call override so the methods would still work https://github.com/facebookincubator/facebook-for-woocommerce/blob/d81f04033cd0ef4db5cfc7c14a1a51cd8f2a58c8/includes/fbproduct.php#L79

you can see this is currently used when passing the facebook product  to this helper method: https://github.com/facebookincubator/facebook-for-woocommerce/blob/a31edf0742b9d13d719f3ac4f98bcda9c0f6bb7c/facebook-commerce.php#L4298

which in turn also calls get_children() here https://github.com/facebookincubator/facebook-for-woocommerce/blob/d81f04033cd0ef4db5cfc7c14a1a51cd8f2a58c8/includes/fbutils.php#L170

I will update the PR since we agreed so far that we can easily switch to the standard WC_Product for the context of these methods

thanks for double checking!